Sources/TSCBasic/HashAlgorithms.swift

```diff
@@ -169,6 +173,25 @@ public struct SHA256: HashAlgorithm {
     }
 }
 
+/// Wraps CryptoKit.SHA256 to provide a HashAlgorithm conformance to it.
+@available(macOS 10.15, *)
+public struct CryptoKitSHA256: HashAlgorithm {
+    public init() {
+    }
+
+    @available(macOS 10.15, *)
+    public func hash(_ bytes: ByteString) -> ByteString {
+        if #available(macOS 10.15, *) {
+            return bytes.withData { data in
+                let digest = CryptoKit.SHA256.hash(data: data)
+                return ByteString(digest)
+            }
+        } else {
+            fatalError("not supported on this platform")
+        }
```

Comment:
I think this will need to use conditional compilation directives.

```diff
-        if #available(macOS 10.15, *) {
+        #if canImport(CryptoKit)
             return bytes.withData { data in
                 let digest = CryptoKit.SHA256.hash(data: data)
                 return ByteString(digest)
             }
-        } else {
+        #else
             fatalError("not supported on this platform")
-        }
+        #endif
```

Reply:
That resolves the Linux issue, but macOS still does not weak link the framework.

Reply:
Someone has reported the weak-linking issue on the [forums][] and as [SR-11907][].

(I'm getting a 502 Proxy Error for bugs.swift.org at the moment.)

[forums]: <https://forums.swift.org/t/weak-linking-of-frameworks-with-greater-deployment-targets/26017/19>
[SR-11907]: <https://bugs.swift.org/browse/SR-11907>